benchmark/Microsoft.IdentityModel.Benchmarks/ReadTokenTests.cs

```diff
@@ -0,0 +1,69 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using BenchmarkDotNet.Attributes;
+using Microsoft.IdentityModel.JsonWebTokens;
+using Microsoft.IdentityModel.Tokens;
+
+namespace Microsoft.IdentityModel.Benchmarks
+{
+    // dotnet run -c release -f net8.0 --filter Microsoft.IdentityModel.Benchmarks.ReadTokenTests*
+
+    [Config(typeof(BenchmarkConfig))]
+    [HideColumns("Type", "Job", "WarmupCount", "LaunchCount")]
+    [MemoryDiagnoser]
+    public class ReadTokenTests
+    {
+        string _encodedJWS;
+        string _encryptedJWE;
+
+        [GlobalSetup]
+        public void Setup()
+        {
+            var jsonWebTokenHandler = new JsonWebTokenHandler();
+            var jwsTokenDescriptor = new SecurityTokenDescriptor
+            {
+                SigningCredentials = BenchmarkUtils.SigningCredentialsRsaSha256,
+                TokenType = JwtHeaderParameterNames.Jwk,
+                Claims = BenchmarkUtils.Claims
+            };
+
+            _encodedJWS = jsonWebTokenHandler.CreateToken(jwsTokenDescriptor);
+
+            var jweTokenDescriptor = new SecurityTokenDescriptor
+            {
+                SigningCredentials = BenchmarkUtils.SigningCredentialsRsaSha256,
+                EncryptingCredentials = BenchmarkUtils.EncryptingCredentialsAes256Sha512,
+                TokenType = JwtHeaderParameterNames.Jwk,
+                Claims = BenchmarkUtils.Claims
+            };
+
+            _encryptedJWE = jsonWebTokenHandler.CreateToken(jweTokenDescriptor);
+        }
+
+        [Benchmark]
+        public JsonWebToken ReadJWS_FromString()
+        {
+            return new JsonWebToken(_encodedJWS);
+        }
+
+        [Benchmark]
+        public JsonWebToken ReadJWS_FromMemory()
+        {
+            return new JsonWebToken(_encodedJWS.AsMemory());
+        }
+
+        [Benchmark]
```

Comment:
```suggestion
        [Benchmark(Baseline = true)]
```

And move it to separate test class

Reply:
Separated JWS and JWE benchmarks.

Results for JWE:
``` ini

BenchmarkDotNet=v0.13.5, OS=Windows 11 (10.0.22631.3007), VM=Hyper-V
AMD EPYC 7763, 1 CPU, 16 logical and 8 physical cores
.NET SDK=8.0.200-preview.23624.5
  [Host]    : .NET 8.0.1 (8.0.123.58001), X64 RyuJIT AVX2
  MediumRun : .NET 8.0.1 (8.0.123.58001), X64 RyuJIT AVX2

Job=MediumRun  MaxAbsoluteError=10.0000 ms  IterationCount=15  
LaunchCount=4  WarmupCount=10  

```
|             Method |     Mean |     Error |    StdDev |   Median |      P90 |      P95 |     P100 |   Gen0 | Allocated |
|------------------- |---------:|----------:|----------:|---------:|---------:|---------:|---------:|-------:|----------:|
| **ReadJWE_FromMemory** | **2.571 μs** | **0.0254 μs** | **0.0557 μs** | **2.547 μs** | **2.667 μs** | **2.671 μs** | **2.681 μs** | **0.1373** |   **2.29 KB** |
| **ReadJWE_FromString** | **2.622 μs** | **0.0183 μs** | **0.0410 μs** | **2.624 μs** | **2.668 μs** | **2.684 μs** | **2.704 μs** | **0.1373** |   **2.29 KB** |

and JWS:
``` ini

BenchmarkDotNet=v0.13.5, OS=Windows 11 (10.0.22631.3007), VM=Hyper-V
AMD EPYC 7763, 1 CPU, 16 logical and 8 physical cores
.NET SDK=8.0.200-preview.23624.5
  [Host]    : .NET 8.0.1 (8.0.123.58001), X64 RyuJIT AVX2
  MediumRun : .NET 8.0.1 (8.0.123.58001), X64 RyuJIT AVX2

Job=MediumRun  MaxAbsoluteError=10.0000 ms  IterationCount=15  
LaunchCount=4  WarmupCount=10  

```
|             Method |     Mean |     Error |    StdDev |   Median |      P90 |      P95 |     P100 | Rank |   Gen0 | Allocated |
|------------------- |---------:|----------:|----------:|---------:|---------:|---------:|---------:|-----:|-------:|----------:|
| **ReadJWS_FromMemory** | **3.276 μs** | **0.0143 μs** | **0.0320 μs** | **3.278 μs** | **3.312 μs** | **3.333 μs** | **3.364 μs** |    **1** | **0.1602** |   **2.67 KB** |
| **ReadJWS_FromString** | **3.253 μs** | **0.0223 μs** | **0.0485 μs** | **3.279 μs** | **3.308 μs** | **3.310 μs** | **3.315 μs** |    **1** | **0.1602** |   **2.67 KB** |